src/higher-order.js

```diff
@@ -152,6 +153,11 @@ function branch(cursors, Component) {
       this.watcher.refresh(mapping);
       this.setState(this.watcher.get());
     }
+
+    // Should update the component?
+    shouldComponentUpdate(nextProps, nextState) {
+      return !(deepEqual(this.props, nextProps, {strict: true}) && deepEqual(this.state, nextState, {strict: true}))
```

Comment:
If you use `===` that won't catch the test case below, since that object that is being set in the tree is different.

I think we should make this function opt-in, (i.e. you have to set an option of "pure=true" for this shouldComponentUpdate to be called)
The other way to accomplish the same thing would be for child components of higher order to have a `shouldComponentUpdate`.